### PR TITLE
feat: add helper to parse `multipart/form-data`

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves https://github.com/unjs/h3/issues/43

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This adds a utility to read `multipart/form-data`: `readMultipartFormData`. It's a separate utility so it doesn't add to the bundle size if not used, as it uses `parse-multipart-data` to handle the parsing.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
Used library seems [a fork](https://github.com/nachomazzara/parse-multipart-data) and [no esm exports](https://unpkg.com/browse/parse-multipart-data@1.5.0/dist/multipart.js). It would be nice if we can inline and cleanup implementation (author shall be be preserved in same file)

**Comment by pi0** [comment]
Can we name it `readFormBody` or `readMultipartBody` for naming consistency?

**Comment by danielroe** [comment]
Shall we just open a PR to add esm exports upstream?

**Comment by pi0** [comment]
I prefer inlining for easier future enhancenments. It is same for other utils.

**Comment by danielroe** [comment]
It just seems less maintainable, that's all, as there's already someone actively maintaining it. But if you wish.

**Comment by pi0** [comment]
Reverted the utility name back to `readMultipartFormData` it was making sense. Also pushed several improvements and bug fixes for ported implementation.

**Review by pi0 — APPROVED** [review]

File: src/utils/multipart.ts [hunk 1]
```diff
@@ -1,42 +1,15 @@
...
-  The above copyright notice and this permission notice shall be included in all
-  copies or substantial portions of the Software.
-
-  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-  SOFTWARE.
```
[the remark this answers is not in the document]
Comment (pi0) [reply src/utils/multipart.ts]: It is tricky if we want to copy paste the same license over everywhere we port code. I don't think it is wrong in any way when as long as license itself and author name is provided and note that the fork we based implementation on, itself broke this and instead added an inline comment :) https://github.com/nachomazzara/parse-multipart-data/blob/master/src/multipart.ts#L2

[the remark this answers is not in the document]
Comment (danielroe) [reply src/utils/multipart.ts]: The fact that the fork didn't abide by the terms of the licence does not mean that we do not need to. It says explicitly that it must be included.

[the remark this answers is not in the document]
Comment (pi0) [reply src/utils/multipart.ts]: Made issue in both to ensure having blessing of both authors :) BTW again, this is not the only code we ported like this, it is not full file, the fork itself removed license which would be probably wrong to even use it (inline or as dep) and the license is still there with links.

**Pull request merged**